src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs

```diff
@@ -409,11 +409,6 @@ public static SyntaxNode CreateNullCheckExpression(this SyntaxGenerator factory,
                     codeFactory.IdentifierName("value")));
             }
 
-            // The user can add required if they want, but if the property being overridden is required, this one needs to be as well.
-            modifiers = modifiers.WithIsRequired(overriddenProperty.IsIndexer
-                ? false
-                : (modifiers.IsRequired || overriddenProperty.IsRequired));
```

Comment:
The indexer part is now done in `CodeGenerationPropertySymbol`, and the required modifier now no longer needs to be special cased here. It'll flow automatically from `GetOverrideModifiers`